spring-boot-3-integrationtests/src/test/java/org/axonframework/utils/InMemoryStreamableEventSource.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (c) 2010-2023. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.utils;
+
+import org.axonframework.common.stream.BlockingStream;
+import org.axonframework.eventhandling.*;
+import org.axonframework.messaging.StreamableMessageSource;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * An in-memory representation of the {@link StreamableMessageSource}. Should be used to test event handling components
+ * requiring test flexibility when events are published.
+ *
+ * @author Allard Buijze
+ * @author Steven van Beelen
+ */
+public class InMemoryStreamableEventSource implements StreamableMessageSource<TrackedEventMessage<?>> {
```

Comment:
Nit: Perhaps we can retrieve this one as a test resource too?

Reply:
Yes, done.